infrastructure/modules/armonik/authentication-in-database.tf

```diff
@@ -0,0 +1,236 @@
+        container {
+          name              = var.authentication.name
+          image             = var.authentication.tag != "" ? "${var.authentication.image}:${var.authentication.tag}" : var.authentication.image
+          image_pull_policy = var.authentication.image_pull_policy
+          command           = ["/bin/bash", "-c", local.authentication_script]
+          env {
+            name  = "MongoDB_Host"
+            value = local.mongodb_host
+          }
+          env {
+            name  = "MongoDB_Port"
+            value = local.mongodb_port
+          }
+          dynamic "env" {
+            for_each = local.pod_authentication_in_database_credentials
+            content {
+              name = env.key
+              value_from {
+                secret_key_ref {
+                  key      = env.value.key
+                  name     = env.value.name
+                  optional = false
+                }
+              }
+            }
+          }
+          dynamic "volume_mount" {
+            for_each = (local.mongodb_certificates_secret != "" ? [1] : [])
+            content {
+              name       = "mongodb-secret-volume"
+              mount_path = "/mongodb"
+              read_only  = true
+            }
+          }
+          volume_mount {
+            name       = "mongodb-script"
+            mount_path = "/mongodb/script"
+            read_only  = true
+          }
+        }
+        dynamic "volume" {
+          for_each = (local.mongodb_certificates_secret != "" ? [1] : [])
+          content {
+            name = "mongodb-secret-volume"
+            secret {
+              secret_name = local.mongodb_certificates_secret
+              optional    = false
+            }
+          }
+        }
+        volume {
+          name = "mongodb-script"
+          config_map {
+            name     = kubernetes_config_map.authmongo.0.metadata[0].name
+            optional = false
+          }
+        }
+      }
+    }
+    backoff_limit = 5
+  }
+  wait_for_completion = true
+  timeouts {
+    create = "2m"
+    update = "2m"
+  }
+}
+
+data "tls_certificate" "certificate_data" {
+  count   = length(tls_locally_signed_cert.ingress_client_certificate)
+  content = tls_locally_signed_cert.ingress_client_certificate[count.index].cert_pem
+}
+
+locals {
+  authentication_data = length(tls_locally_signed_cert.ingress_client_certificate) > 0 ? jsonencode({
+    certificates_list = [for index, cert in data.tls_certificate.certificate_data : {
+      "Fingerprint" = cert.certificates[length(cert.certificates) - 1].sha1_fingerprint
+      "CN"          = tls_cert_request.ingress_client_cert_request[index].subject.0.common_name
+      "Username"    = local.ingress_generated_cert.names[index]
+    }]
+    users_list = [for index, cert in data.tls_certificate.certificate_data : {
+      "Username" = local.ingress_generated_cert.names[index],
+      "Roles"    = [local.ingress_generated_cert.names[index]]
```

Comment:
Do Username and Role have the same value ?

Reply:
Only because they are auto-generated, otherwise they can be different. It was just a simplification in this case